mantle/kola/tests/ignition/kdump.go

```diff
@@ -0,0 +1,222 @@
+        Flags:       []register.Flag{},
+        Distros:     []string{"rhcos", "fcos"},
+        Tags:        []string{"kdump", kola.NeedsInternetTag, "reprovision", kola.SkipBaseChecksTag},
+    })
+}
+
+// TangServer contains fields required to set up a tang server
+type SshServer struct {
+    Machine           platform.Machine
+    MachineAddress    string
+    SSHPort           string
+    PrivSSH           string
+    PubSSH            string
+}
+
+
+// Start a VM and wreturn the SSH key pair.
+func setupSSHMachine(c cluster.TestCluster) SshServer {
+    var m platform.Machine
+    var err error
+    var address string
+    var port string
+
+    options := platform.QemuMachineOptions{
+        HostForwardPorts: []platform.HostForwardPort{
+            {Service: "ssh", HostPort: 0, GuestPort: 22},
+        },
+    }
+
+    // temp dir to store SSH keys
+    tmpd, err := os.MkdirTemp("", "kola-kdump-crash-ssh")
+    if err != nil {
+        c.Fatalf("Error creating tempdir: %v", err)
+    }
+    defer os.RemoveAll(tmpd)
+
+    // generate an ssh key pair we'll use for authentication
+    pubkeyBuf, privkeyPath, err := util.CreateSSHAuthorizedKey(tmpd)
+    if err != nil {
+        c.Fatalf("Error creating ssh keys: %v", err)
+    }
+
+    // load the private key as well
+    PrivKeyBuf, err := os.ReadFile(privkeyPath)
+    if err != nil {
+          c.Fatalf("error reading pubkey: %v", err)
+    }
+
+
+    // Inject the public key previously created as an
+    // authorized key
+    ignition := conf.Ignition(fmt.Sprintf(`{
+       "ignition": { "version": "3.4.0" },
+       "passwd":{
+         "users":[
+           {
+             "name":"core",
+             "sshAuthorizedKeys":["%s"]
+            }
+          ]
+       }
+    }`, strings.TrimSpace(string(pubkeyBuf))))
+
+
+    // get machine address
+    switch pc := c.Cluster.(type) {
+        // These cases have to be separated because when put together to the same case statement
+        // the golang compiler no longer checks that the individual types in the case have the
+        // NewMachineWithQemuOptions function, but rather whether platform.Cluster
+        // does which fails
+        case *qemu.Cluster:
+            m, err = pc.NewMachineWithQemuOptions(ignition, options)
+            for _, hfp := range options.HostForwardPorts {
+                if hfp.Service == "ssh" {
+                    address = "10.0.2.2"
+                    port = fmt.Sprintf("%d",hfp.HostPort)
+                }
+            }
+        default:
+            m, err = pc.NewMachine(ignition)
+            address = m.IP()
+            port = "22"
+        }
+        if err != nil {
+            c.Fatal(err)
+        }
+
+    // insert indentation in front of the entries
+    // to avoid errors in the butane file
+    var padded = ""
+    for _, line := range strings.Split(strings.TrimSuffix(string(PrivKeyBuf), "\n"), "\n") {
+        padded = fmt.Sprintf("%s          %s\n", padded, line)
+    }
```

Comment:
Let's put this hack at least in `kdumpSSHTest` instead since that's where the indentation level is defined.